pkg/querier/querier_test.go

```diff
@@ -228,6 +228,93 @@ func TestNoHistoricalQueryToIngester(t *testing.T) {
 
 }
 
+func TestNoFutureQueries(t *testing.T) {
+	testCases := []struct {
+		name               string
+		mint, maxt         time.Time
+		hitStores          bool
+		maxQueryIntoFuture time.Duration
+	}{
+		{
+			name:               "hit-test1",
+			mint:               time.Now().Add(-5 * time.Hour),
+			maxt:               time.Now().Add(1 * time.Hour),
+			hitStores:          true,
+			maxQueryIntoFuture: 10 * time.Minute,
+		},
+		{
+			name:               "hit-test2",
+			mint:               time.Now().Add(-5 * time.Hour),
+			maxt:               time.Now().Add(-59 * time.Minute),
+			hitStores:          true,
+			maxQueryIntoFuture: 10 * time.Minute,
+		},
+		{ // Skipping stores is disabled.
+			name:               "hit-test2",
+			mint:               time.Now().Add(-5 * time.Hour),
+			maxt:               time.Now().Add(-50 * time.Minute),
+			hitStores:          true,
+			maxQueryIntoFuture: 0,
+		},
+		{ // Still hit because of staleness.
+			name:               "hit-test3",
+			mint:               time.Now().Add(12 * time.Minute),
+			maxt:               time.Now().Add(60 * time.Minute),
+			hitStores:          true,
+			maxQueryIntoFuture: 10 * time.Minute,
+		},
+		{
+			name:               "dont-hit-test1",
+			mint:               time.Now().Add(100 * time.Minute),
+			maxt:               time.Now().Add(5 * time.Hour),
+			hitStores:          false,
+			maxQueryIntoFuture: 10 * time.Minute,
+		},
+		{
+			name:               "dont-hit-test2",
+			mint:               time.Now().Add(16 * time.Minute),
+			maxt:               time.Now().Add(60 * time.Minute),
+			hitStores:          false,
+			maxQueryIntoFuture: 10 * time.Minute,
+		},
+	}
+
+	engine := promql.NewEngine(promql.EngineOpts{
+		Logger:        util.Logger,
+		MaxConcurrent: 10,
+		MaxSamples:    1e6,
+		Timeout:       1 * time.Minute,
+	})
+	cfg := Config{}
```

Comment:
It's always a good practice to initialize it with defaults. You can just do `flagext.DefaultValues(&cfg)`.